src/containers/PDiskPage/DecommissionButton/DecommissionButton.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+
+import {ChevronDown} from '@gravity-ui/icons';
+import type {DropdownMenuItem, DropdownMenuProps} from '@gravity-ui/uikit';
+import {Button, DropdownMenu, Icon, Popover} from '@gravity-ui/uikit';
+
+import {CriticalActionDialog} from '../../../components/CriticalActionDialog/CriticalActionDialog';
+import {isErrorWithRetry} from '../../../components/CriticalActionDialog/utils';
+import type {EDecommitStatus} from '../../../types/api/pdisk';
+import {wait} from '../../../utils';
+import {cn} from '../../../utils/cn';
+import {pDiskPageKeyset} from '../i18n';
+
+import decommissionIcon from '../../../assets/icons/decommission.svg';
+
+import './DecommissionButton.scss';
+
+const b = cn('ydb-pdisk-decommission-button');
+
+function getDecommissionWarningText(decommission?: EDecommitStatus) {
+    if (decommission === 'DECOMMIT_IMMINENT') {
+        return pDiskPageKeyset('decommission-dialog-imminent-warning');
+    }
+    if (decommission === 'DECOMMIT_PENDING') {
+        return pDiskPageKeyset('decommission-dialog-pending-warning');
+    }
+    if (decommission === 'DECOMMIT_REJECTED') {
+        return pDiskPageKeyset('decommission-dialog-rejected-warning');
+    }
+    if (decommission === 'DECOMMIT_NONE') {
+        return pDiskPageKeyset('decommission-dialog-none-warning');
+    }
+
+    return undefined;
+}
+
+interface DecommissionButtonProps {
+    decommission?: EDecommitStatus;
+    onConfirmAction: (newDecomissionStatus?: EDecommitStatus, isRetry?: boolean) => Promise<void>;
+    onConfirmActionSuccess: (() => Promise<void>) | VoidFunction;
+    buttonDisabled?: boolean;
+    popoverDisabled?: boolean;
+}
+
+export function DecommissionButton({
+    decommission,
+    onConfirmAction,
+    onConfirmActionSuccess,
+    buttonDisabled,
+    popoverDisabled,
+}: DecommissionButtonProps) {
+    const [newDecommission, setNewDecommission] = React.useState<EDecommitStatus | undefined>();
+    const [buttonLoading, setButtonLoading] = React.useState(false);
+    const [withRetry, setWithRetry] = React.useState(false);
+
+    const handleConfirmAction = async (isRetry?: boolean) => {
+        setButtonLoading(true);
+        await onConfirmAction(newDecommission, isRetry);
```

Comment:
This doesn't handle errors properly. If onConfirmAction throws an error, setButtonLoading(false) will never be called, potentially leaving the button in a perpetual loading state. Wrap this in a try/catch and ensure setButtonLoading(false) is always called.

Reply:
It's wrapped inside `CriticalActionDialog`, `handleConfirmActionSuccess` and `handleConfirmActionError` are called on success and error respectively. But you are right, there is no need to `setButtonLoading(false)` here, since it's done in other handlers